### PR TITLE
ACCUMULO-3734 unescape auths when using AuthsContainer

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: core/src/test/java/org/apache/accumulo/core/security/VisibilityEvaluatorTest.java [hunk 1]
```diff
@@ -113,6 +123,31 @@ public void testQuote() {
...
+  public void testUnescape() {
+    assertEquals("a\"b", VisibilityEvaluator.unescape(new ArrayByteSequence("a\\\"b")).toString());
+    assertEquals("a\\b", VisibilityEvaluator.unescape(new ArrayByteSequence("a\\\\b")).toString());
+    assertEquals("a\\\"b", VisibilityEvaluator.unescape(new ArrayByteSequence("a\\\\\\\"b")).toString());
+    assertEquals("\\\"", VisibilityEvaluator.unescape(new ArrayByteSequence("\\\\\\\"")).toString());
+    assertEquals("a\\b\\c\\d", VisibilityEvaluator.unescape(new ArrayByteSequence("a\\\\b\\\\c\\\\d")).toString());
+
+    try {
+      VisibilityEvaluator.unescape(new ArrayByteSequence("a\\b"));
+      fail();
```
Comment (joshelser) [inline core/src/test/java/org/apache/accumulo/core/security/VisibilityEvaluatorTest.java]: An assertion message for the `fail()`'s would be nice.

File: core/src/main/java/org/apache/accumulo/core/security/VisibilityEvaluator.java [hunk 2]
```diff
@@ -26,6 +28,58 @@
 public class VisibilityEvaluator {
   private AuthorizationContainer auths;
 
+  private static class UnescapingAuthorizationContainer implements AuthorizationContainer {
```
Comment (joshelser) [inline core/src/main/java/org/apache/accumulo/core/security/VisibilityEvaluator.java]: Javadoc would be nice

**Comment by joshelser** [comment]
Minor nits, otherwise LGTM

**Comment by keith-turner** [comment]
@joshelser  thanks for looking at it.   I plan on pulling this back to 1.6, merging it forward,  and pushing it to apache Mon morning.

**Pull request closed without merging**